doc/source/examples/cfrl_mnist.ipynb

```diff
@@ -40,10 +40,10 @@
     "import tensorflow.keras as keras\n",
```

Comment:
Now that tensorflow will not be installed by default, do we want to add notes to the example notebooks to say that pip install alibi[tensorflow] must be run in order for the notebook to be installed? (currently the `import tensorflow...` etc won't actually work with after a generic `pip install alibi`). 

This could also be put at a higher level in the docs if we went for something more general like pip install -r requirements/examples.txt...

Reply:
Would it be too much work to split this is PR in two stages (or squash the appropriate commits) - 1. general framework 2. making `tensorflow` optional?

Reply:
w.r.t the notebooks installing the optional dependencies I've added a Note informing the user they may need to do so. This is in keeping with the previous approach taken for the shap example notebooks. The only place I haven't done so is in the case where the dependency is independent of the alibi. So in the anchor image examples, they import tensorflow to train the classifier but the `AnchorImage` implementation doesn't use tensorflow explicitly itself.